### PR TITLE
Bastion host dns

### PR DESCRIPTION
**Pull request by jschummer**
This change is [Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27)

### CODE REVIEW

**Review by pallost — CHANGES_REQUESTED** [review]
Looks pretty good to me, but would need some minor refactoring

> Reviewed 1 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-:-MXf4bnj2HtFLbOJllzf:bhw5ele)* status: 1 of 5 files reviewed, 3 unresolved discussions (waiting on @jschummer)

___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 15 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3qhU9sPVQ0mNDNL1:-MXf3qhU9sPVQ0mNDNL2:b-e02302) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L15)):*
> ```YAML
>     msg: "{{ public_ip.stdout }}"
> 
> - set_fact: bastion_host_dns_name="bastion.{{ subdomain }}"
> ```

does this notation work correctly with the prod domain, i.e. "bastion.@.jore.hsl.fi" will become "bastion.jore.hsl.fi" as expected?
___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 22 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3aopACGq1KHgYn_O:-MXf3aopACGq1KHgYn_P:bp1z6uy) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L22)):*
> ```YAML
>     resource_group: "{{ az_project_name }}-dns"
>     location: "{{ az_region_name }}"
>     name: "azure_dns_zones"
> ```

this deployment should be named something else, e.g. "bastion_host_dns_zone"
___
*[ansible/roles/azure_dns_zone/tasks/main.yml, line 39 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf48lk0WrRtWmsji7f:-MXf48lk0WrRtWmsji7g:b-69d174) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/main.yml#L39)):*
> ```YAML
>     template: "{{ lookup('file', 'templates/dnszone.arm.json') }}"
> 
> - include_tasks: bastion-host.yml
> ```

would rather arrange this role then like this:
- tasks/env-zone.yml -> creates the dns zone for the environment (*.jore.hsl.fi)
- tasks/bastion-host-zone.yml -> creates the dns zone for the bastion host (bastion.*.jore.hsl.fi)
- tasks/main.yml -> includes the previous two

**Review by jschummer — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27)* status: 1 of 6 files reviewed, 3 unresolved discussions (waiting on @pallost)

___
*[ansible/roles/azure_dns_zone/tasks/main.yml, line 39 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf48lk0WrRtWmsji7f:-MXhZOxu5pfdlal43ln7:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/main.yml#L39)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

would rather arrange this role then like this:
- tasks/env-zone.yml -> creates the dns zone for the environment (*.jore.hsl.fi)
- tasks/bastion-host-zone.yml -> creates the dns zone for the bastion host (bastion.*.jore.hsl.fi)
- tasks/main.yml -> includes the previous two
</blockquote></details>

Done.
___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 15 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3qhU9sPVQ0mNDNL1:-MXhYFRLBcilrGR0i0ZL:b-8c3r3t) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L15)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

does this notation work correctly with the prod domain, i.e. "bastion.@.jore.hsl.fi" will become "bastion.jore.hsl.fi" as expected?
</blockquote></details>

Not really :-) But that's why this step is only run when bastion_host_dns_name is not set (for prod it is set manually, so this is skipped)

Could / should I maybe add a comment about that? (Where / how?)
___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 22 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3aopACGq1KHgYn_O:-MXhYm102NJr5xzOJJj_:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L22)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

this deployment should be named something else, e.g. "bastion_host_dns_zone"
</blockquote></details>

Done.

**Review by jschummer — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27)* status: 1 of 6 files reviewed, 2 unresolved discussions (waiting on @pallost)

___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 15 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3qhU9sPVQ0mNDNL1:-MXl00haAh1EBNcJMbqx:b-dw63zm) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L15)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

well why don't you just create a new variable for each `env-*.yaml`? Like `bastion_subdomain=bastion.dev` in DEV or `bastion_subdomain=bastion` in PROD
</blockquote></details>

The idea was that this follows the same scheme for all envs except for prod. So this way we'd say 'it follows the rule dns_name="bastion." + env' except if you want to override it explicitly (which we do want to do for prod) and this way would save some duplicate "code" (i.e. variable assignments).

Is it causing more confusion than helping understand the code?

**Review by pallost — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27)* status: 1 of 6 files reviewed, 1 unresolved discussion (waiting on @jschummer)

___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 15 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3qhU9sPVQ0mNDNL1:-MY4G7at3nsCK4_xZLG6:b-wvbhr1) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L15)):*
<details><summary><i>Previously, jschummer wrote…</i></summary><blockquote>

The idea was that this follows the same scheme for all envs except for prod. So this way we'd say 'it follows the rule dns_name="bastion." + env' except if you want to override it explicitly (which we do want to do for prod) and this way would save some duplicate "code" (i.e. variable assignments).

Is it causing more confusion than helping understand the code?
</blockquote></details>

yeah, it makes is hard to follow that some envs do define the `bastion_host_dns_name` variable and some don't :) You could just do the same assignments in your env-* files:
```
bastion_host_dns_name="bastion.{{ subdomain }}" # in the env-dev.yml
bastion_host_dns_name="bastion.{{ subdomain }}" # in the env-test.yml
bastion_host_dns_name="bastion" # in the env-prod.yml
```
which would make it clear that the bastion dns name is based on the subdomain

**Review by jschummer — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27)* status: 1 of 9 files reviewed, 1 unresolved discussion (waiting on @pallost)

___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 15 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3qhU9sPVQ0mNDNL1:-MY4uvr76a8wvefXrRcj:bdj41pr) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L15)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

yeah, it makes is hard to follow that some envs do define the `bastion_host_dns_name` variable and some don't :) You could just do the same assignments in your env-* files:
```
bastion_host_dns_name="bastion.{{ subdomain }}" # in the env-dev.yml
bastion_host_dns_name="bastion.{{ subdomain }}" # in the env-test.yml
bastion_host_dns_name="bastion" # in the env-prod.yml
```
which would make it clear that the bastion dns name is based on the subdomain
</blockquote></details>

Alright, you convinced me :-) Since 1/3 is special cases this is probably really the clearer way.

**Comment by jschummer** [comment]
Fixes jore4#83

**Review by pallost — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27)* status: 1 of 9 files reviewed, 1 unresolved discussion (waiting on @jschummer)

___
*[ansible/roles/azure_dns_zone/tasks/bastion-host.yml, line 15 at r1](https://reviewable.io/reviews/hsldevcom/jore4-deploy/27#-MXf3qhU9sPVQ0mNDNL1:-MYENSuW9_50vwvXk7WZ:bn2g1ef) ([raw file](https://github.com/hsldevcom/jore4-deploy/blob/6cb62d54e4a04ac03c6b14b6cfc610b68c3b8e17/ansible/roles/azure_dns_zone/tasks/bastion-host.yml#L15)):*
<details><summary><i>Previously, jschummer wrote…</i></summary><blockquote>

Alright, you convinced me :-) Since 1/3 is special cases this is probably really the clearer way.
</blockquote></details>

good that I managed to convince you :) also, this will make it easier to use the bastion host hostname is a parameter in other playbooks
